### PR TITLE
enh(ceip): add agent configuration to ceip stats

### PR DESCRIPTION
**Pull request by a-launois**
## Description
Add  poller/agent configuration stat to CEIP cron:
- total of configurations
- total of configurations per type
- total of pollers per type
```
{
    ...
    "poller-agent-configuration": {
        "total": int,
        "telegraf": { 
            "configuration": int, 
            "pollers": int
        },
       ...
    }
}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

### CODE REVIEW

File: centreon/www/class/centreonStatistics.class.php [hunk 1]
```diff
@@ -407,6 +407,48 @@ public function getAccData(): array
...
+        $data = ['total' => 0];
+
+        $statement = $this->dbConfig->executeQuery(
+            <<<'SQL'
+                SELECT count(id)
+                FROM agent_configuration ac
+                SQL
+        );
+
+        $total = $statement->fetchColumn();
```
Comment (Drago37) [inline centreon/www/class/centreonStatistics.class.php:426]: Now, with CentreonDB, you must not to use PDO native, but the metods of CentreonDB ;) 
```suggestion
        $total = $this->dbConfig->fetchColumn($statement);
```

File: centreon/www/class/centreonStatistics.class.php [hunk 2]
```diff
@@ -407,6 +407,48 @@ public function getAccData(): array
...
+                JOIN ac_poller_relation rel
+                    ON ac.id = rel.ac_id
+                GROUP BY type
+                SQL
+        );
+
+        foreach ($statement->fetchAll(\PDO::FETCH_ASSOC) as $row) {
+            $data[$row['type']]['configuration'] = $row['nb_ac_per_type'];
+            $data[$row['type']]['pollers'] = $row['nb_poller_per_type'];
+        }
```
Comment (Drago37) [inline centreon/www/class/centreonStatistics.class.php:447]: Now, with CentreonDB, you must not to use PDO native, but the metods of CentreonDB ;) 
Also, a variable to store the result of fetchAll before the loop is more optimized :)
```suggestion
        $acAndPollerPerType = $this->dbConfig->fetchAll($statement);
        foreach ($acAndPollerPerType as $row) {
            $data[$row['type']]['configuration'] = $row['nb_ac_per_type'];
            $data[$row['type']]['pollers'] = $row['nb_poller_per_type'];
        }
```

File: centreon/www/class/centreonStatistics.class.php [hunk 3]
```diff
@@ -407,6 +407,48 @@ public function getAccData(): array
...
+
+        $total = $statement->fetchColumn();
+
+        if ($total === false) {
+            return $data;
+        }
+
+        $data['total'] = $total;
+
+        $statement = $this->dbConfig->executeQuery(
```
Comment (Drago37) [inline centreon/www/class/centreonStatistics.class.php:434]: Please note that the method doesn't have the @throw tag in her php docs to alert she can throw a CentreonDbException ;)

**Review by Drago37 — CHANGES_REQUESTED** [review]

File: centreon/www/class/centreonStatistics.class.php (hunk 3 above)
Comment (Drago37) [reply centreon/www/class/centreonStatistics.class.php:436]: good !

**Review by Drago37 — APPROVED** [review]

File: centreon/www/class/centreonStatistics.class.php (hunk 1 above)
Comment (Drago37) [reply centreon/www/class/centreonStatistics.class.php]: ok

File: centreon/www/class/centreonStatistics.class.php (hunk 2 above)
Comment (Drago37) [reply centreon/www/class/centreonStatistics.class.php]: ok

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**